Website/src/rendering/src/redux/ocCurrentCart/index.ts

```diff
@@ -207,7 +207,30 @@ export const createLineItem = createOcAsyncThunk<RequiredDeep<DOrderWorksheet>,
       const orderResponse = await Orders.Create<DOrder>('All', { xp: { DeliveryType: 'Ship' } });
       orderId = orderResponse.ID;
     }
-    await LineItems.Create<DLineItem>('All', orderId, request);
+
+    // Determine if the line item is already in the cart
```

Comment:
With these changes, the app is failing to compile in the build pipeline with this error:
```
Failed to compile.

./src/redux/ocCurrentCart/index.ts:453:9
Type error: Type 'RequiredObjectDeep<LineItem<DLineItemXp, DProductXp, DVariantXp, DShippingAddressXp, DShipFromAddressXp>>[]' is not assignable to type 'WritableDraft<RequiredObjectDeep<DLineItem>>[]'.
  Type 'RequiredObjectDeep<LineItem<DLineItemXp, DProductXp, DVariantXp, DShippingAddressXp, DShipFromAddressXp>>' is not assignable to type 'WritableDraft<RequiredObjectDeep<DLineItem>>'.
    The types of 'ShipFromAddress.xp' are incompatible between these types.
      Type 'RequiredObjectDeep<DShipFromAddressXp>' is missing the following properties from type 'WritableDraft<RequiredObjectDeep<DShipFromAddress>>': ID, DateCreated, CompanyName, FirstName, and 10 more.

  451 |       if (action.payload) {
  452 |         state.order = action.payload.Order;
> 453 |         state.lineItems = action.payload.LineItems;
      |         ^
  454 |         state.shippingAddress = state.lineItems?.length ? state.lineItems[0].ShippingAddress : null;
  455 |         state.shipEstimateResponse = action.payload.ShipEstimateResponse;
  456 |         state.initialized = true;

```

Reply:
Thanks, should be fixed now. I also just added a change to make sure we are always setting the spec values on URL changes, otherwise products without specs were getting the previous product's specs included in the add to cart payload.